gqlalchemy/memgraph.py

```diff
@@ -586,3 +586,45 @@ def to_cypher_lambda(expression: str) -> str:
             return ""
         else:
             return f"(r, n | {expression})"
+
+
+class WeightedShortestPath(IntegratedAlgorithm):
+    """Build a Djikstra shortest path call for a Cypher query
+
+    The weighted shortest path algorithm can be called in Memgraph with Cypher
+    queries such as:
+    " MATCH (a {id: 723})-[r *WSHORTEST 10 (r, n | r.weight) weight_sum
+            (r, n | r.x > 12 AND r.y < 3)]-(b {id: 882}) RETURN * "
+    It is called inside the relationship clause, "*WSHORTEST" naming the
+    algorithm, "10" specifying search depth bounds, and "(r, n | <expression>)"
+    is a filter lambda, used to filter which relationships and nodes to use.
+    """
+
+    def __init__(
+        self,
+        upper_bound: int = None,
+        condition: str = None,
+        total_weight_var: str = "total_weight",
+        weight_property: str = "r.weight",
+    ) -> None:
+        super().__init__()
+        if "." not in weight_property:
+            self.weight_property = "r." + weight_property
+        else:
+            self.weight_property = weight_property
```

Comment:
Use ternary operator again.
self.weight_property = f"r.{weight_property}" if "." not in weight_property else weight_property